dev.md

```diff
@@ -1,36 +1,121 @@
+1. sudo apt-get update
+2. sudo apt-get install -y gcc-9 g++-9
+3. sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 800 --slave /usr/bin/g++ g++ /usr/bin/g++-9
+4. (Optional) check version: `gcc -v`
+
+
+### Install CURL, UNWIND, IBERTY
+1. sudo apt-get install -y libcurl4-gnutls-dev
+2. sudo apt-get install -y libunwind-dev
+3. sudo apt-get install -y libiberty-dev
+
+### Install DOUBLE-CONVERSION
+1. git clone https://github.com/google/double-conversion.git
+2. cd double-conversion
+3. cmake -DBUILD_SHARED_LIBS=OFF . && make && sudo make install
+
+### Install GFLAGS, GLOG, GTEST
+1. sudo apt-get install -y libgflags-dev
+2. git clone https://github.com/google/glog.git
+3. cd glog && make . && sudo make install
+1. git clone https://github.com/google/googletest.git
+2. mkdir googletest/build && cd googletest/build
+3. cmake .. && make && sudo make install
+
+### install MBEDTLS
+1. git clone https://github.com/ARMmbed/mbedtls.git
+2. mkdir mbedtls/build && cd mbedtls/build
+3. cmake -DCMAKE_CXX_FLAGS=-fPIC -DCMAKE_C_FLAGS=-fPIC .. && make -j$(nproc)
+4. sudo make install
+
+### Install LIBEVENT
+1. git clone https://github.com/libevent/libevent.git
+2. cd libevent
+3. cmake . && make -j$(nproc)
+4. sudo make install
+
+### Install FMT
+1. git clone https://github.com/fmtlib/fmt.git
+2. cd fmt
+3. cmake . && make -j$(nproc)
+4. sudo make install
+
+### Install BOOST
+1. sudo apt-get install -y libboost-dev
+
+### Install Facebook Folly
+1. git clone https://github.com/facebook/folly.git
+2. mkdir _build && cd _build
+3. cmake ..
+4. make -j$(nproc)
+5. sudo make install
+
+### Install compression libs
+1. sudo apt-get install -y liblz4-dev
+2. sudo apt-get install -y libzstd-dev
+3. sudo apt-get install -y libsnappy-dev
+4. sudo apt-get install -y liblzma-dev
+
+### Install OpenSSL
+1. git clone https://github.com/openssl/openssl.git
+2. cd openssl
+3. ./config
+4. make -j$(nproc)
+5. sudo make install
+
+### Install gperftools
+1. sudo apt-get install -y autoconf
+2. git clone https://github.com/gperftools/gperftools.git
+3. cd gperftools && ./autogen.sh && ./configure && sudo make install
+
+### Install protobuf
+1. git clone https://github.com/protocolbuffers/protobuf.git
+2. cd protobuf && git submodule update --init --recursive && ./autogen.sh
+3. ./configure && make -j$(nproc) && sudo make install && sudo ldconfig
+
+### fix grpc
+A local change to solve the the build break (in `~/nebula/build`):
+1. grpc/src/grpc/include/grpcpp/impl/codegen/byte_buffer.h
+   move method to public grpc_byte_buffer* c_buffer() { return buffer_; }
+2. flatbuffers/src/flatbuffers/include/flatbuffers/grpc.h
+   use ByteBuffer as parameter type of Deserialize
+     + static grpc::Status Deserialize(ByteBuffer *bb, flatbuffers::grpc::Message<T> *msg) {
+     + grpc_byte_buffer* buffer = nullptr;
+     + if (!bb || !(buffer = bb->c_buffer())) {
+         return ::grpc::Status(::grpc::StatusCode::INTERNAL, "No payload");
+       }
```

Comment:
How about move this section under “Install linker dependencies”, it is better to finish all the decencies first, and the modified the code.

Reply:
Actually, maybe under build nebula, after cmake ..

Reply:
Added one more step before this to ensure both flatbuffers and grpc downloaded.